jwst/cube_build/ifu_cube.py

```diff
@@ -1809,6 +1878,21 @@ def map_nirspec_pixel_to_sky(self, input_model):
         x, y, ra, dec, lambda, slice_no
 
         """
+
+        # check if we have an ra and dec offset file
+        raoffset = 0.0
+        decoffset = 0.0 
+        # pull out ra dec offset if it exists
+        if offsets is not None:
+            filename = input_model.meta.filename
+            index = offsets['filename'].index(filename)
+            raoffset = offsets['raoffset'][index]
+            decoffset = offsets['decoffset'][index]
+            log.info("Ra and dec offset (arc seconds) applied to file :%5.2f, %5.2f,  %s",
+                     raoffset, decoffset, filename)
+            raoffset = raoffset/3600.0 # convert to degress
+            decoffset = decoffset/3600.0
```

Comment:
Lines 1882 - 1894 are repeated three times. Might be better to have a function.

Reply:
pulled out modifying units to after finding median dec. So it is not just in one place.